File: personal_project_back-end/common-service/src/main/java/com/example/common/service/models/RoomModel.java
package com.example.common.service.models;
import com.example.common.service.enums.RoomStatusEnum;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class RoomModel extends BaseModel {
    @NotBlank(message = "Room number is required")
    private String number;

    private Double price = 0.0;

    private Double length;
    private Double width;

    @NotNull(message = "Status is required")
    private RoomStatusEnum status;

    private String note;
}

File: personal_project_front-end/src/models/RoomModel.ts
import { RoomStatusEnum } from "@/enums/RoomStatusEnum";
import { BaseModel } from "./Basemodel";

export type RoomModel = BaseModel & {
  id?: number;
  price: number;
  number: string;
  length?: number;
  width?: number;
  status: RoomStatusEnum;
  note?: string;
};

export const defaultRoomModel: RoomModel = {
  price: 0,
  number: "",
  length: 1,
  width: 1,
  status: RoomStatusEnum.AVAILABLE,
  note: "",
};
